src/logic/wallets/ethTransactions.ts

```diff
@@ -38,18 +40,21 @@ export const calculateGasPrice = async (): Promise<string> => {
   return Promise.reject(err)
 }
 
-export const calculateGasOf = async (txConfig: {
-  to: string
-  from: string
-  data: string
-  gasPrice?: number
-  gas?: number
-}): Promise<number> => {
-  const web3 = getWeb3()
+export const calculateGasOf = async (txConfig: EthAdapterTransaction): Promise<number> => {
   try {
-    const gas = await web3.eth.estimateGas(txConfig)
+    const web3 = getWeb3()
+    const signerAddress = await getAccountFrom(web3)
+
+    if (!signerAddress) {
+      throw new CodedException(Errors._103, 'signerAddress returned null')
+    }
+
+    const ethAdapter = new Web3Adapter({
```

Comment:
It looks good to me, but I think at some point we should have only one instance in the app that is used in multiple places. Same with the `safeSdk` once it is used.

Reply:
Makes sense. I'll move it now so we don't have to in the future

Reply:
Will leave this be for the moment as `signerAddress` requires `string`, whereas our `getAddressFrom()` can return `string | null`. 

@germartinez will let it accept null and throw in methods where it's used. `estimateGas` does not use it, so this is currently not an issue.